hazelcast-client/src/main/resources/hazelcast-client-full-example.xml

```diff
@@ -0,0 +1,879 @@
+        Hazelcast XML file right after the <hazelcast-client> element, and it should have the required attribute "resource".
+
+        Below is an example where the configuration files you want to include are located at your
+        Hazelcast working directory:
+        <import resource="your-client-configuration-file.xml"/>
+
+        Below are examples showing a classpath or filesystem location:
+        <import resource="file:///etc/hazelcast/your-client-configuration-file-1.xml"/>
+        <import resource="classpath:your-client-configuration-file-2.xml"/>
+
+        Below is an example showing property placeholders:
+        <import resource="${environment}-your-client-configuration-file.xml"/>
+
+    -->
+    <import resource="your-client-configuration-XML-file"/>
+
+    <!--
+        The <config-replacers> allow to use variables (placeholders) within the configuration file and use an external
+        class to retrieve correct values (replacements).
+
+        It can be used for masking sensitive strings such as passwords for instance.
+
+        Format of a variable is:
+        $ PREFIX { STRING_TO_BE_REPLACED }      e.g. $ENC{nnPgTqJCcCQ=:23000:B4y/nlp6M0t3q6YiKImW+w==}
+
+        The PREFIX value depends on the replacer implementation (e.g. "ENC" is used for the EncryptionReplacer)
+        The STRING_TO_BE_REPLACED is the value which is provided to replacer implementation.
+    -->
+    <config-replacers fail-if-value-missing="false">
+        <replacer class-name="com.hazelcast.config.replacer.EncryptionReplacer">
+            <properties>
+                <property name="passwordFile">password.txt</property>
+                <property name="passwordUserProperties">false</property>
+                <property name="cipherAlgorithm">DES</property>
+                <property name="keyLengthBits">64</property>
+                <property name="secretKeyAlgorithm">DES</property>
+                <property name="secretKeyFactoryAlgorithm">PBKDF2WithHmacSHA1</property>
+            </properties>
+        </replacer>
+    </config-replacers>
+
+    <!--
+        Specifies the name and password for a cluster group you connect to.
+        Cluster groups allow you to create separate sub-clusters with different names.
+        The password is only checked when security is enabled on Hazelcast members (Enterprise edition feature).
+    -->
+    <group>
+        <name>dev</name>
+        <password>dev-pass</password>
+    </group>
+
+    <!--
+        When Hazelcast instances are created, they are put in a global registry with their creation names.
+        <instance-name> elements gives you the ability to get a specific Hazelcast instance from this registry
+        by giving the instance's name.
+    -->
+    <instance-name>CLIENT_NAME</instance-name>
+
+    <!--
+        ===== HAZELCAST LICENSE CONFIGURATION =====
+
+        If you have an Enterprise or Enterprise HD license, you can enter it here.
```

Comment:
On the client, licence is not required. This config is for backward compatibility. We may update the comment to imply that it's not required.